src/signals/incident/components/form/Header/index.js

```diff
@@ -1,58 +1,80 @@
-        <span className="header--not-required">(optioneel)</span>
-      : ''}
-    </div>
-    <div className="header__subtitle">{meta && meta.subtitle}</div>
 
-    <div className="header__errors">
-      <div className="header__errors__item">
-        {touched
-        && hasError('required')
-        && 'Dit is een verplicht veld'}
-      </div>
+const Header = ({ meta, options, touched, hasError, getError, children, parent }) => {
+  const {
+    label,
+    subtitle,
+  } = meta;
 
-      <div className="header__errors__item">
-        {touched
-        && hasError('email')
-        && 'Het moet een geldig e-mailadres zijn'}
-      </div>
+  const incident = get(parent, 'meta.incident', {});
+  const shallowValues = flattenObject(incident, '', ':');
+  const isLabelDescriptor = label !== null && typeof label === 'object';
 
-      <div className="header__errors__item">
-        {touched
-        && hasError('maxLength')
-        && `U kunt maximaal ${getError('maxLength').requiredLength} tekens invoeren.`}
+  return (
+    <div className={`header ${touched && (hasError('required') || hasError('email') || hasError('maxLength') || hasError('custom')) ? 'header--invalid' : ''}`}>
+      <div className="header__label">{
+        isLabelDescriptor ?
+          <FormattedMessage {...label} values={shallowValues} />
+          : label
+        }
+        {(label && (!options || !options.validators)) || (options && options.validators && !options.validators.includes(Validators.required)) ?
+          <span className="header--not-required">(optioneel)</span>
+          : ''}
       </div>
+      <div className="header__subtitle">{subtitle}</div>
+
+      <div className="header__errors">
+        <div className="header__errors__item">
+          {touched
+          && hasError('required')
+          && 'Dit is een verplicht veld'}
+        </div>
 
-      <div className="header__errors__item">
-        {touched
-        && hasError('custom')
-        && getError('custom')}
+        <div className="header__errors__item">
+          {touched
+          && hasError('email')
+          && 'Het moet een geldig e-mailadres zijn'}
+        </div>
+
+        <div className="header__errors__item">
+          {touched
+          && hasError('maxLength')
+          && `U kunt maximaal ${getError('maxLength').requiredLength} tekens invoeren.`}
+        </div>
+
+        <div className="header__errors__item">
+          {touched
+          && hasError('custom')
+          && getError('custom')}
+        </div>
       </div>
-    </div>
 
-    <div className="header__children">
-      {children}
+      <div className="header__children">
+        {children}
+      </div>
     </div>
-  </div>
-);
+  );
+};
 
 Header.propTypes = {
-  meta: PropTypes.object,
+  meta: PropTypes.shape({
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
```

Comment:
Using `PropTypes.object` should fail; discouraged and I believe even disallowed by the linter configuration. Use `PropTypes.shape({})` instead and define the props that you expect to receive.